notebooks/Notebook3.md

```diff
@@ -18,17 +20,72 @@ exciting milestones, preliminary results, etc. Feel free to include images
+
+New route: Just have one broad for-loop structure. Instead of having any overly complicated set of control structures, I just decided to create one broad for loop. "for (i = 0, i < x; i++)" style for loops can allow you to do essentially any kind of repition, and that is what I decided to add into the language. Some overloading allows the for loop to be even easier and more intuitive to be used. 
+
+I had already made a simple for-loop in class, as part of the puzzle. So, I built something similar this time. There are two key differences though. Firstly, the overloading. I used overloading to have loops with defaults. You can choose to define all three (i = 0, i < 10, i += 1), or you can have two (i < 10, i += 1) where the default is just "i = 0", or you can have (10) where the default is "i = 0 and i += 1" (single increment, with 0 indexing). I thought about the third option being (i < 10), but since most people would use "i <" each time, I decided to abstract that way and make things easer for the programmer by just having (10). They don't have to use this option, but since most of the remaining boiler plate code of for-loops is the same, "for(10)" is a convenient option to have. The user need not remember multiple loop keywords since all of them go under the same name. Because, overloading.
+
+The second difference is the syntax. I used 2 parameter lists this time, instead of four. They are parameter LISTS, so I just had the first one have all three, as a list. (i = 0, i < 10, i += 1) {body}. (Instead of for_loop (i = 0) (i < 10) (i += 1) (body).)
+
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+I am actually happy with the overall direction. Both weeks, I added a lot of operations, functions, abstractions, conversions, control structures, fluency, and DSL-yness and I would like to continue moving in this direction. I still have to add more functions, operators, DSLy-ness, control structures - to actually land at the desired goal of a well designed DSL. I have a good basis in terms of implementation and design to work on and, eventually, meet my goal.
+
+My main pressing issue though, is that it still feels like a general language and not a DSL. But, as I keep changing and bending Scala each week - little by little, I hope it will be more like a DSL by the end.
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+What are some changes I can make to remove the flavor of the host language - Scala? In many ways, it already feels different. But in many ways, it is still feels like Scala with a lot of garnish on top. I was wondering what I could do to make it more domain specific and take out any non-essential Scala-ness from the language.
+
```

Comment:
Are there some specific pieces of syntax that still feel to Scala-y? Can you point us to an example and describe what is not ideal (and potentially what you would like to say)?